libvirt/container-linux/kubernetes/variables.tf

```diff
@@ -0,0 +1,146 @@
+  description = "The domain name to use for the cluster (e.g. k8s.example.com)"
+}
+
+variable "k8s_hostname" {
+  description = "hostname (NOT FQDN) which will be used to point to a controller"
+  type        = "string"
+  default     = "console"
+}
+
+# machines
+
+variable "ssh_authorized_key" {
+  type        = "string"
+  description = "SSH public key for user 'core'"
+}
+
+variable "asset_dir" {
+  description = "Path to a directory where generated assets should be placed (contains secrets)"
+  type        = "string"
+}
+
+# controllers
+# The controller_ips must be the same length as controller_names
+variable "controller_names" {
+  description = "list of hostnames to use for the controllers. Each machine will have the domain appende"
+  type        = "list"
+}
+
+variable "controller_ips" {
+  description = "list of static IPs to use for the controllers"
+  type        = "list"
+}
+
+variable "controller_memory" {
+  description = "ram to allocate in MiB for each controller"
+  type        = "string"
+  default     = "2048"
+}
+
+# workers
+# worker IPs are assigned via DHCP
+variable "worker_names" {
+  type = "list"
+}
+
+variable "worker_memory" {
+  description = "ram to allocate in MiB for each worker"
+  type        = "string"
+  default     = "2048"
+}
+
+
+# Optional cluster networking configuration
+
+variable "networking" {
+  description = "Choice of networking provider (flannel or calico)"
+  type        = "string"
+  default     = "calico"
+}
+
+variable "network_mtu" {
+  description = "CNI interface MTU (applies to calico only)"
+  type        = "string"
+  default     = "1480"
+}
+
+variable "network_ip_autodetection_method" {
+  description = "Method to autodetect the host IPv4 address (applies to calico only)"
+  type        = "string"
+  default     = "first-found"
+}
+
+variable "pod_cidr" {
+  description = "CIDR IPv4 range to assign Kubernetes pods"
+  type        = "string"
+  default     = "10.2.0.0/16"
+}
+
+variable "service_cidr" {
+  description = <<EOD
+CIDR IPv4 range to assign Kubernetes services.
+The 1st IP will be reserved for kube_apiserver, the 10th IP will be reserved for coredns.
+EOD
+
+  type    = "string"
+  default = "10.3.0.0/16"
+}
+
+# Optional libvirt configuration
+
+variable "node_ip_pool" {
+  description = "the IP pool to use for machine DHCP"
+  type        = "string"
+  default     = "192.168.120.0/24"
+}
+
+variable "libvirt_network_name" {
+  description = "The name for the libvirt network object. If empty, use the cluster name"
+  type        = "string"
+  default     = ""
+}
+
+variable "libvirt_network_if" {
+  description = "The name for the host network interface"
+  type        = "string"
+  default     = "br42"
+}
+
+variable "libvirt_storage_pool" {
+  description = "The libvirt storage pool to use"
+  type        = "string"
+  default     = "default"
+}
+
+variable "libvirt_endpoint" {
+  type    = "string"
+  default = "qemu:///system"
+}
+
+variable "dns_server" {
```

Comment:
Remove

Reply:
We do need something like this; libvirt requires an upstream DNS server.